### DISCUSSION TITLE
Chance to see BabylonNative in NativeScript?

### DISCUSSION THREAD

**Discussion by cjohn001** [BabylonJS/BabylonNative#1139] [Development]
Hello together,
I am wondering, if there is a chance to see BabylonNative support in NativeScript in the near future?

**Comment by bghgary** [comment]
Sorry for the delay response. I had missed this one.

I don't know if we have much experience with NativeScript, so I can't say one way or another. Do you have any thoughts on how they would integrate?

**Comment by cjohn001** [comment]
Hello @bghgary,
thanks for the reply. Regarding the integration I need to say that I am not very familiar with native programming on android and ios. However, I currently see two possibilities.

1)

In the meantime there is a new plugin available:

https://github.com/NativeScript/canvas/tree/feat/v2

which already integrates a 4.x version of Babylon (v2 branch of the plugin in the github repo, no official release yet). 

To my understanding the people developing the plugin, have  written a low level wrapper to execute opengl from nativescript and this wrapper is used from babylon already. 
However, the project seems to be very young and I have not had success yet to execute babylon5 libraries with the plugin yet. I think it might be possible that your implementation is already more optimised for babylon specifics. However, I cannot really rate this assumption. So option one would mean, just upgrade babylon to version 5 on the canvas plugin and all is set already.

2) Option 2 is something which might be interesting to discuss with the maintainers of the canvas plugin. Maybe your low level code could be placed into the plugin in some way. I think triniwiz  is one of the experts and main developers of the plugin.

**Reply by bghgary** [reply]
The tech sounds similar to expo.

Both approaches sound like they hook into webgl in some way, which is very different from what Babylon Native does, which hooks in at the Babylon.js engine level. I think it will be difficult to integrate.
